### PR TITLE
`Development`: Use signals in lecture unit wizard

### PR DESCRIPTION
**Pull request by florian-glombik**
#### General
- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/development-process.html#naming-conventions-for-github-pull-requests).

#### Client
- [x] I **strictly** followed the [client coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-tests/).
- [x] I added multiple screenshots/screencasts of my UI changes.

### Motivation and Context
This PR introduces signals, and fixes code smells in components used for editing and updating lectures. This is a preparation for https://github.com/ls1intum/Artemis/pull/9655 and aims to keep https://github.com/ls1intum/Artemis/pull/9655 at a manageable size of changes.

### Description
* Introduced signals in `lecture-attachments.component.ts` (also refined the usage of lifecycle hooks there)
* Introduced signals in `lecture-wizard-units.component.ts`
* Adjusted client tests to work with signals

### Steps for Testing
Prerequisites:
- 1 Instructor
- 1 Lecture

1. Create a new lecture attachment
2. Verify the lecture attachment is saved properly
3. Create a new lecture unit
4. Verify the lecture unit is saved properly
5. Edit the lecture unit
6. Verify there are no issues _(that are not present on develop aswell)_ with editing the lecture unit

[](https://artemis-test1.artemis.cit.tum.de)
[](https://artemis-test2.artemis.cit.tum.de)
[](https://artemis-test3.artemis.cit.tum.de)
[](https://artemis-test4.artemis.cit.tum.de)
[](https://artemis-test5.artemis.cit.tum.de)
[](https://artemis-test6.artemis.cit.tum.de)
[](https://artemis-test9.artemis.cit.tum.de)

#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [ ] Test 1
- [ ] Test 2

### Screenshots
lectureUnitSignalsRefactoring

## Summary by CodeRabbit

- **New Features**
	- Enhanced reactivity for unit form visibility in the lecture wizard.
	- Improved conditional rendering for unit forms.
	- Updated property access in the lecture attachments component for better functionality.

- **Bug Fixes**
	- Corrected logic for displaying the appropriate forms based on user interactions.

- **Chores**
	- Removed empty constructors from components to simplify code structure.
	- Refined test cases for better input handling and error management in lecture attachments.

### CODE REVIEW

File: src/main/webapp/app/lecture/lecture-attachments.component.ts [hunk 1]
```diff
@@ -46,30 +51,31 @@ export class LectureAttachmentsComponent implements OnInit, OnDestroy {
...
-                this.lectureService.findWithDetails(this.lectureId).subscribe((lectureResponse: HttpResponse<Lecture>) => {
-                    this.lecture = lectureResponse.body!;
-                    this.loadAttachments();
+    private routeDataSubscription?: Subscription;
+
+    constructor() {
+        effect(
+            () => {
+                this.notificationText = undefined;
+                this.routeDataSubscription?.unsubscribe(); // in case the subscription was already defined
```
Comment (JohannesStoehr) [inline src/main/webapp/app/lecture/lecture-attachments.component.ts:60]: How could it be defined before the constructor?

Comment (florian-glombik) [reply src/main/webapp/app/lecture/lecture-attachments.component.ts:60]: right now it is within an effect, so if `lectureId()` would change, we would overwrite the previous subscription and loose track if the pervious one (potentially causing memory leaks)

It _should_ never happen though

**Review by JohannesStoehr — APPROVED** [review]
Code

**Review by BBesrour — APPROVED** [review]
code lgtm

**Review by magaupp — APPROVED** [review]
Tested TS3. Lecture attachments and lecture unit editing showed no problems

**Review by coolchock — APPROVED** [review]
tested in testing session, works as expected

**Review by HawKhiem — APPROVED** [review]
Tested on TS1. Creating and editing attachments as well as units works fine

**Pull request merged**